src/scripts/main.js

```diff
@@ -1 +1,86 @@
 'use strict';
+
+const showMessage = (text, className) => {
+  const message = document.createElement('div');
+
+  message.setAttribute('data-qa', 'notification');
+  message.textContent = text;
+  message.className = `message ${className}`;
+  document.body.appendChild(message);
+};
+
+document.addEventListener('contextmenu', (ev) => {
+  ev.preventDefault();
+});
+
+const firstPromise = new Promise((resolve, reject) => {
+  let isSettled = false;
+
+  const clickHandler = () => {
+    if (!isSettled) {
+      isSettled = true;
+      resolve('First promise was resolved');
+      document.removeEventListener('click', clickHandler);
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  setTimeout(() => {
+    if (!isSettled) {
+      isSettled = true;
+      reject(new Error('First promise was rejected'));
+      document.removeEventListener('click', clickHandler);
+    }
+  }, 3000);
+});
+
+firstPromise
+  .then((message) => {
+    showMessage(message, 'success');
+  })
+  .catch((error) => {
+    showMessage(error.message, 'error');
+  });
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('mousedown', function handler(ev) {
+    if (ev.button === 0) {
+      resolve('Second promise was resolved on left click');
+      document.removeEventListener('mousedown', handler);
+    } else if (ev.button === 2) {
+      resolve('Second promise was resolved on right click');
+      document.removeEventListener('mousedown', handler);
+    }
+  });
+});
+
+secondPromise.then((message) => {
+  showMessage(message, 'success');
+});
+
+let leftClick = false;
+let rightClick = false;
```

Comment:
The variables `leftClick` and `rightClick` are used to track mouse button clicks for the third promise. However, they are not reset after the promise resolves. This means if the user clicks both buttons again, the promise will not resolve a second time unless the page is reloaded. Consider resetting these variables after the promise resolves if you want to allow multiple resolutions.